packages/@react-aria/landmark/src/useLandmark.ts

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaLabelingProps} from '@react-types/shared';
+import {HTMLAttributes, MutableRefObject, useCallback, useEffect, useState} from 'react';
+import {useLayoutEffect} from '@react-aria/utils';
+
+export type AriaLandmarkRole = 'main' | 'region' | 'search' | 'navigation' | 'form' | 'banner' | 'contentinfo' | 'complementary';
+
+export interface AriaLandmarkProps extends AriaLabelingProps {
+  role: AriaLandmarkRole
+}
+
+interface LandmarkAria {
+  landmarkProps: HTMLAttributes<HTMLElement>
+}
+
+type Landmark = {
+  ref: MutableRefObject<HTMLElement>,
+  role: AriaLandmarkRole,
+  label?: string,
+  lastFocused?: HTMLElement,
+  onFocus: () => void,
+  onBlur: () => void
```

Comment:
nit: if these are methods rather than event handlers, they should be named `focus()` and `blur()`.